lib/public_ip/service/simple.rb

```diff
@@ -1,12 +1,29 @@
 require 'net/http'
 require 'timeout'
 
+IPV4_REGEX = /(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])/
+
 module PublicIp
   module Service
+    class InvalidIpAddress < StandardError; end
+    class TimedOut < StandardError; end
+
     class Simple
       attr_reader :uri
       attr_reader :headers
 
+      def self.ip
+        response = perform_request
+
+        ip_address = extract_ip(response)
+
+        unless IPV4_REGEX.match(ip_address)
```

Comment:
@nhinds That is interesting, I'll try to broaden the regex to allow both IPv4 and IPv6 addresses.

Thanks for the review 👍